--- a/psplash-colors.h	2017-01-17 15:51:42.855468000 +0100
+++ b/psplash-colors.h	2017-01-17 13:36:42.726216345 +0100
@@ -20,7 +20,7 @@
 #define _HAVE_PSPLASH_COLORS_H
 
 /* This is the overall background color */
-#define PSPLASH_BACKGROUND_COLOR 0xec,0xec,0xe1
+#define PSPLASH_BACKGROUND_COLOR 0x00,0x00,0x00
 
 /* This is the color of any text output */
 #define PSPLASH_TEXT_COLOR 0x6d,0x6d,0x70
@@ -29,6 +29,6 @@
 #define PSPLASH_BAR_COLOR 0x6d,0x6d,0x70
 
 /* This is the color of the progress bar background */
-#define PSPLASH_BAR_BACKGROUND_COLOR 0xec,0xec,0xe1
+#define PSPLASH_BAR_BACKGROUND_COLOR 0x00,0x00,0x00
 
 #endif
